enterprise/server/cmd/executor/executor.go

```diff
@@ -293,7 +294,21 @@ func main() {
 	if err != nil {
 		log.Fatalf("Error initializing executor registration: %s", err)
 	}
-	reg.Start(rootContext)
+
+	warmupDone := make(chan struct{})
+	go func() {
+		executionServer.Warmup()
+		close(warmupDone)
+	}()
+	go func() {
+		if executorConfig.StartupWarmupMaxWaitSecs != 0 {
+			select {
+			case <-warmupDone:
+			case <-time.After(time.Duration(executorConfig.StartupWarmupMaxWaitSecs) * time.Second):
```

Comment:
what happens in this case?

Reply:
We allow the startup to proceed without waiting any longer.